### PR TITLE
feat: EU component Icon

### PR DESCRIPTION
**Pull request by nimek2**

### CODE REVIEW

**Comment by hapertown** [comment]
deploy/netlify failed - the problem is probably on the environment configuration side. Do you not have to configure the webpack.config.js like [here](https://github.com/ec-europa/ecl-twig/blob/master/src/ec/.storybook/webpack.config.js) file for the EU in the same way as for the EC file to correctly display the svg file?

File: src/eu/packages/eu-component-icon/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "name": "@ecl-twig/eu-component-icon",
+  "author": "European Commission",
+  "license": "EUPL-1.1",
+  "version": "0.0.1-alpha",
+  "description": "ECL Twig - EU Icon",
+  "publishConfig": {
+    "access": "public"
+  },
+  "devDependencies": {
+    "@ecl/ec-resources-icons": "^2.0.0"
```
Comment (planctus) [inline src/eu/packages/eu-component-icon/package.json]: @nimek2 , the thing here is that you are providing a component as part of an ecosystem, in this case is EU so your resources should come from the EU implementation of ECL in this case, and all of your imports then would use the resources provided by this package, not the EC one.. ;)

For the rest yes, the problem about the icons not rendering is due to webpack conf, i will fix that, once you fix this EC-EU confusion the pull request will be merged, we already reviewed the EC version.
Thanks!

Comment (hapertown) [reply src/eu/packages/eu-component-icon/package.json]: I changed and commited the resource from ec-resources-icons to eu-resources-icons, is it enough?

File: src/eu/packages/eu-component-icon/icon.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import path from 'path';
+import { renderTwigFile } from '@ecl-twig/test-utils';
+
+import brandedIcons from '@ecl/ec-resources-icons/dist/lists/branded.json';
```
Comment (planctus) [inline src/eu/packages/eu-component-icon/icon.test.js]: Now that we are fetching the eu package for the resources we should use the icons from that one, so here we would import from the EU package and not the EC one.

Comment (hapertown) [reply src/eu/packages/eu-component-icon/icon.test.js]: Fixed in [b49cf0a
](https://github.com/ec-europa/ecl-twig/pull/15/commits/b49cf0a7c4788b648f51be03431c9254f3956ace) commit

File: src/ec/packages/ec-component-icon/icon.story.js [hunk 3]
```diff
@@ -3,11 +3,11 @@ import { withKnobs, select } from '@storybook/addon-knobs';
 import { withNotes } from '@ecl-twig/storybook-addon-notes';
 import withCode from '@ecl-twig/storybook-addon-code';
 
-import brandedIcons from '@ecl/ec-resources-icons/dist/lists/branded.json';
-import generalIcons from '@ecl/ec-resources-icons/dist/lists/general.json';
-import notificationsIcons from '@ecl/ec-resources-icons/dist/lists/notifications.json';
-import uiIcons from '@ecl/ec-resources-icons/dist/lists/ui.json';
-import defaultSprite from '@ecl/ec-resources-icons/dist/sprites/icons.svg';
+import brandedIcons from '@ecl/eu-resources-icons/dist/lists/branded.json';
```
Comment (emeryro) [inline src/ec/packages/ec-component-icon/icon.story.js]: I guess there is a confusion between EC and EU here...
EC components (as this file), should use EC dependencies, no EU, and the same for the other way

File: src/eu/packages/eu-component-icon/docs/icon.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# ECL Twig - EC Icon component
```
Comment (emeryro) [inline src/eu/packages/eu-component-icon/docs/icon.md]: there should be no reference to EC in this file (including npm package below, and so on)

File: src/eu/packages/eu-component-icon/icon.story.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
+import { storiesOf } from '@storybook/html';
+import { withKnobs, select } from '@storybook/addon-knobs';
+import { withNotes } from '@ecl-twig/storybook-addon-notes';
+import withCode from '@ecl-twig/storybook-addon-code';
+
+import brandedIcons from '@ecl/ec-resources-icons/dist/lists/branded.json';
```
Comment (emeryro) [inline src/eu/packages/eu-component-icon/icon.story.js]: should be EU everywhere here too

File: src/eu/packages/eu-component-icon/icon.test.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import generalIcons from '@ecl/ec-resources-icons/dist/lists/general.json';
+import notificationsIcons from '@ecl/ec-resources-icons/dist/lists/notifications.json';
```
Comment (emeryro) [inline src/eu/packages/eu-component-icon/icon.test.js]: should be EU everywhere here too

**Comment by hapertown** [comment]
I will pay more attention to the correct paths of resources for next PRs, so that we will not repeat this problem, thanks for your help.

**Comment by emeryro** [comment]
Almost there.
Please just make sure to use EC in EC related components (https://github.com/ec-europa/ecl-twig/pull/15#discussion_r255511600)

File: src/ec/packages/ec-component-icon/icon.story.js (hunk 3 above)
Comment (nimek2) [reply src/ec/packages/ec-component-icon/icon.story.js]: Fixed here https://github.com/ec-europa/ecl-twig/pull/15/commits/7b43a8964e4a47f6a5d930518de753c95058232d

That was stupid mistake

**Comment by nimek2** [comment]
@emeryro 

All fixes done (including focusable parameter)
Auto test passed so can we merge it now?

**Comment by nimek2** [comment]
@emeryro 

Aria-hidden parameter added. SO now we are good here?

**Pull request merged**